devicetypes/smartthings/arrival-sensor-ha.src/arrival-sensor-ha.groovy

```diff
@@ -79,21 +100,37 @@ def beep() {
 }
 
 def parse(String description) {
-    state.lastCheckin = now()
-    handlePresenceEvent(true)
-
-    if (description?.startsWith('read attr -')) {
-        handleReportAttributeMessage(description)
+	log.debug "description: $description"
+	state.lastCheckin = now()
+    if (device.getDataValue("manufacturer") == "SmartThings"){
+    	handlePresenceEvent(true)
     }
-
-    return []
-}
-
-private handleReportAttributeMessage(String description) {
-    def descMap = zigbee.parseDescriptionAsMap(description)
-    if (descMap.clusterInt == 0x0001 && descMap.attrInt == 0x0020) {
-        handleBatteryEvent(Integer.parseInt(descMap.value, 16))
+    
+    if ((description?.startsWith("catchall:")) || (description?.startsWith("read attr -"))) {
+    	log.debug zigbee.parseDescriptionAsMap(description)
+    	def descMap = zigbee.parseDescriptionAsMap(description)
+        if (device.getDataValue("manufacturer") == "Vision" && descMap && descMap.clusterInt == 0x0006) {
+            log.debug "Command: ${descMap.commandInt}"
+        	if(descMap.commandInt == 0x01)
+            {
+            	log.debug "True"
+				handlePresenceEventVision(true)	
```

Comment:
Why did You add new method: handlePresenceEventVision ? Since it's almost identical to handlePresenceEvent ,You may add some if-statements in handlePresenceEvent to handle Your device's events.

Reply:
OK! I will add some if-statements in handlePresenceEvent to handle my device's events.